### PR TITLE
test(synchronized): add base tests for two chart tooltip/cursor sync

### PR DESCRIPTION
**Pull request by ckifer**
## Description

Write some base tests for chart synchronization. Very very similar to other existing tooltip tests

## Related Issue

https://github.com/recharts/recharts/issues/4548

## Motivation and Context

write tests so we don't break things
https://github.com/recharts/recharts/issues/4548

## How Has This Been Tested?

 npm test

## Types of changes

- [ ] Bug fix (non-breaking change which fixes an issue)
- [ ] New feature (non-breaking change which adds functionality)
- [ ] Breaking change (fix or feature that would cause existing functionality to change)
- [X] tests

## Checklist:

- [ ] My change requires a change to the documentation.
- [ ] I have updated the documentation accordingly.
- [X] I have added tests to cover my changes.
- [ ] I have added a storybook story or extended an existing story to show my changes

### CODE REVIEW

File: test/component/Tooltip/Tooltip.sync.spec.tsx [hunk 1]
```diff
@@ -0,0 +1,382 @@
...
+          <Wrapper syncId="tooltipSync">
+            <Tooltip />
+          </Wrapper>
+        </>,
+      );
+      const wrappers = container.querySelectorAll('.recharts-wrapper');
+      const wrapperTwo = wrappers[1];
+
+      // target the first chart to show the tooltip
+      showTooltip(container, mouseHoverSelector, debug);
```
Comment (PavelVanecek) [inline test/component/Tooltip/Tooltip.sync.spec.tsx:291]: How do we know this targets the first chart? Should it instead do `showTooltip[wrappers[0], ...`?

File: test/component/Tooltip/Tooltip.sync.spec.tsx [hunk 2]
```diff
@@ -0,0 +1,382 @@
...
+
+      // target the second chart to see if it has the synchonized tooltip showing
+      const tooltip = getTooltip(wrapperTwo);
+      expect(tooltip).toBeVisible();
+
+      const tooltipContentName = wrapperTwo.querySelector('.recharts-tooltip-item-name');
+      const tooltipContentValue = wrapperTwo.querySelector('.recharts-tooltip-item-value');
+      expect(tooltipContentName).not.toBeNull();
+      expect(tooltipContentValue).not.toBeNull();
+      expect(tooltipContentName).toBeInTheDocument();
```
Comment (PavelVanecek) [inline test/component/Tooltip/Tooltip.sync.spec.tsx:301]: Can we also `expect`:

- the tooltip content (name + value)
- same for the other tooltip? both of them should display, right? and each with different content depending on their respective `dataKey`s

**Review by PavelVanecek — APPROVED** [review]
Nice, a great start. The tooltip sync support is worse than I expected. Something with potential to improve in 3.2 or somewhere.

I have some comments but let's follow up on those in the next PR, and merge this!

File: test/component/Tooltip/Tooltip.sync.spec.tsx (hunk 1 above)
Comment (ckifer) [reply test/component/Tooltip/Tooltip.sync.spec.tsx:291]: Yeah probably

File: test/component/Tooltip/Tooltip.sync.spec.tsx (hunk 2 above)
Comment (ckifer) [reply test/component/Tooltip/Tooltip.sync.spec.tsx:301]: Sure. And probably different content given different data, but the same indices

**Pull request merged**